spec/serializers/user_bookmark_serializer_spec.rb

```diff
@@ -33,6 +33,19 @@
     expect(s.post_user_avatar_template).not_to eq(nil)
   end
 
+  it "uses the correct highest_post_number column based on whether the user is staff" do
+    Fabricate(:post, topic: bookmark.topic)
+    Fabricate(:post, topic: bookmark.topic)
+    Fabricate(:whisper, topic: bookmark.topic)
+    list = BookmarkQuery.new(user: bookmark.user).list_all.to_ary
+    s = UserBookmarkSerializer.new(list.last, scope: Guardian.new(user))
+    expect(s.highest_post_number).to eq(3)
+    user.update!(admin: true)
+    list = BookmarkQuery.new(user: bookmark.user).list_all.to_ary
+    s = UserBookmarkSerializer.new(list.last, scope: Guardian.new(user))
+    expect(s.highest_post_number).to eq(4)
```

Comment:
```suggestion
    list = BookmarkQuery.new(user: bookmark.user).list_all.to_ary
    serializer = UserBookmarkSerializer.new(list.last, scope: Guardian.new(user))
    
    expect(serializer.highest_post_number).to eq(3)
    
    user.update!(admin: true)
    serializer = UserBookmarkSerializer.new(list.last, scope: Guardian.new(user))
    
    expect(serializer.highest_post_number).to eq(4)
```

I suspect this will work so we don't have to query for things twice.